Source/santactl/Commands/SNTCommandStatus.mm

```diff
@@ -33,6 +33,38 @@
   }
 }
 
+NSString *Plural(uint64_t val, NSString *base) {
+  if (val == 1) {
+    return base;
+  } else {
+    return [base stringByAppendingString:@"s"];
+  }
+}
+
+NSString *FormatTimeRemaining(uint64_t seconds) {
+  if (seconds < 60) {
+    return @"Less than 1 minute remaining";
+  }
+
+  // Round to nearest minute
+  uint64_t minutes = (seconds + 30) / 60;
+  uint64_t hours = minutes / 60;
+  uint64_t remainingMinutes = minutes % 60;
+
+  if (hours > 0) {
+    if (remainingMinutes > 0) {
+      return [NSString stringWithFormat:@"About %llu %@ and %llu %@ remaining", hours,
+                                        Plural(hours, @"hour"), remainingMinutes,
+                                        Plural(remainingMinutes, @"minute")];
+    } else {
+      return [NSString stringWithFormat:@"About %llu %@ remaining", hours, Plural(hours, @"hour")];
+    }
+  } else {
+    return
+        [NSString stringWithFormat:@"About %llu %@ remaining", minutes, Plural(minutes, @"minute")];
+  }
+}
```

Comment:
Can't we just use  [localizedStringFromTimeInterval]( https://developer.apple.com/documentation/foundation/relativedatetimeformatter/localizedstring(fromtimeinterval:)?language=objc)

e.g. Something like

```C
- (NSString *)FormatTimeRemaining:(NSTimeInterval)seconds {
    NSDateComponentsFormatter *formatter = [[NSDateComponentsFormatter alloc] init];
    formatter.unitsStyle = NSDateComponentsFormatterUnitsStyleFull;
    formatter.allowedUnits = NSCalendarUnitDay | NSCalendarUnitHour | 
                             NSCalendarUnitMinute | NSCalendarUnitSecond;
    formatter.maximumUnitCount = 2;
    formatter.collapsesLargestUnit = NO;
    
    return [formatter stringFromTimeInterval:seconds];
}
```

Reply:
Neat! didn't know about this API.